File: src/main/java/it/unibo/crabinv/Model/entities/bullets/EnemyBulletFactory.java
package it.unibo.crabinv.Model.entities.bullets;

public class EnemyBulletFactory  implements BulletFactory{
    private final double ENEMY_BULLET_RADIUS = 0.01;
    private final double ENEMY_BULLET_SPEED = 0.01;
    private final int ENEMY_BULLET_MAX_HEALTH = 1;
    @Override
    public Bullet createBullet(double x, double y, double minBound, double maxBound) {
        return new BulletEnemy(ENEMY_BULLET_MAX_HEALTH, x,y,ENEMY_BULLET_RADIUS,ENEMY_BULLET_SPEED,minBound,maxBound);
    }
}

File: src/main/java/it/unibo/crabinv/model/entities/bullets/EnemyBulletFactory.kt
package it.unibo.crabinv.model.entities.bullets

/**
 * It's the enemyBulletFactory that handles it.
 */
class EnemyBulletFactory : BulletFactory {
    override fun createBullet(
        x: Double,
        y: Double,
        minBound: Double,
        maxBound: Double,
    ): Bullet =
        BulletEnemy(
            ENEMY_BULLET_MAX_HEALTH,
            x,
            y,
            ENEMY_BULLET_RADIUS,
            ENEMY_BULLET_SPEED,
            minBound,
            maxBound,
        )

    companion object {
        private const val ENEMY_BULLET_RADIUS = 0.01
        private const val ENEMY_BULLET_SPEED = 0.01
        private const val ENEMY_BULLET_MAX_HEALTH = 1
    }
}
